### PR TITLE
gpsbabel 1.7.0

### PR DESCRIPTION
**Pull request by tsteven4**
- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----
update version to latest.
correct dependencies.  On macOS gpsbabel 1.7.0 does use an included version of libusb-1.0, but the necessary objects are compiled from source included in the gpsbabel package and linked into the gpsbabel executable.  'otool -L gpsbabel' will not show a dependency on libusb.

### CODE REVIEW

**Comment by tsteven4** [comment]
The existence of the ubuntu test makes me question the change I made to the dependencies.  While libusb-1.0 is not a dependency on macOS, it is a dependency on linux.  I am unsure of how to make this distinction in the formula.

**Comment by SMillerDev** [comment]
Can we force it to use libusb?

**Comment by tsteven4** [comment]
On macOS: there is not a configure option to use a system libusb, gpsbabel is always built and linked with the included libusb source.

On linux: there is a configure option to build without a libusb dependency, --without-libusb.  However, this will result in reduced functionality.  There is not a configure option to build with included libusb source.

Also note that previous to the gpsbabel 1.7.0 release libusb-0.1 was used.  gpsbabel 1.7.0 is the first release to use libusb-1.0.

**Comment by SMillerDev** [comment]
I would really prefer if you found a way to have macOS use the system/homebrew libusb. There's no real reason for the difference as far as I can tell.

**Comment by tsteven4** [comment]
Using the system/homebrew libusb library on macOS is not supported by any upstream build process.

**Comment by SMillerDev** [comment]
Could you make a pull request to enable that, because compiling software that has a formula is heavily discouraged. It brings a host of potential security and compatibility issues.

**Comment by tsteven4** [comment]
upstream is unlikely to add customization options for library location, similar recent requests have not been accepted.  While configure is still present in the upstream repository, it's use is discouraged.  how about a patch like the following.
```
+    on_macos do
+      # on macOS force use of homebrew libusb-1.0 instead of included version.
+      inreplace "GPSBabel.pro", %r{^.*mac/libusb.*$}, ""
+      system "qmake", "GPSBabel.pro", "CONFIG+=link_pkgconfig", "PKGCONFIG+=libusb-1.0"
+    end
+    on_linux do
+      # on linux system libusb-1.0 is used by default
+      system "qmake", "GPSBabel.pro"
+    end
+    system "make"
+    bin.install "gpsbabel"
```

File: Formula/gpsbabel.rb
Comment (SMillerDev) [inline Formula/gpsbabel.rb:22]: Can you link one of the issues from upstream about using system libusb on Mac here?

Comment (SeekingMeaning) [reply Formula/gpsbabel.rb:22]: Wouldn't this fall under https://docs.brew.sh/Formula-Cookbook#inreplace

> `inreplace` should be used instead of patches when patching something that will never be accepted upstream, e.g. making the software’s build system respect Homebrew’s installation hierarchy.

The next sentence seems to imply that a patch/PR/issue should be made only if it doesn't build on macOS:

> If it’s something that affects both Homebrew and MacPorts (i.e. macOS specific) it should be turned into an upstream submitted patch instead.

Comment (SMillerDev) [reply Formula/gpsbabel.rb:22]: If it's a couple of lines we can just use inreplace. If it's larger we can't really avoid patches.

You should always submit pull requests for anything that's broken because it's the right thing to do 😁. I think the documentation here is incompleted compared to our intent to carry as little patches as possible.

Comment (tsteven4) [reply Formula/gpsbabel.rb:22]: As one of the upstream maintainers I can tell you we would reject the patch corresponding to the inreplace lines because they are too specific.  It is often unexpectedly difficult to simultaneously meet the needs of all the distributions, including our own packaging for windows and macOS available at gpsbabel.org, and not overburden the code base.  Almost all distributions currently use specific targeted patches to bridge the gap between what is supplied upstream and their desires.

What would be useful is if someone who is familiar with homebrew's policies and desires could open an issue on https://github.com/gpsbabel/gpsbabel.git stating your desire and motivation, e.g. "It is homebrews policy to use ... because ... .  Can you please provide a method to support this?"  Having a homebrew representative open the request will make it possible for the upstream maintainers to ask further questions.  I warn you that I don't think a resolution will be forthcoming in the short term, we have been contemplating how to reduce the number of build systems we support.

Comment (SMillerDev) [reply Formula/gpsbabel.rb:22]: Done: https://github.com/gpsbabel/gpsbabel/issues/608

Comment (SMillerDev) [reply Formula/gpsbabel.rb:22]: @tsteven4 you mentioned "upstream is unlikely to add customization options for library location, similar recent requests have not been accepted.", could you link one of the similar requests here?

Comment (tsteven4) [reply Formula/gpsbabel.rb:22]: Here is a PR that attempted to add configurability for shapelib: https://github.com/gpsbabel/gpsbabel/pull/387
Here is a new PR that adds configurability to the qmake flow for zlib/shapelib/libusb: https://github.com/gpsbabel/gpsbabel/pull/611

BTW homebrew doesn't have pkg-config support for shapelib.

Comment (tsteven4) [reply Formula/gpsbabel.rb:22]: The lack of pkg-config for shapelib is due to https://github.com/OSGeo/shapelib/issues/13, although presumably this could be added in the shapelib formula, or shapelib formula could use configure instead of cmake.

**Comment by tsteven4** [comment]
Ok, we have agreement upstream to support configuration for shapelib, zlib, libusb-1.0 with the qmake flow.  I will probably make some additional changes before we merge the upstream PR.

1. I assume you want to use homebrew libusb-1.0 and shapelib, and macOS zlib.  Is that correct?
2. without pkgconfig support for shapelib I need to know the include and lib path for shapelib.  I see some formula referencing 
#{Formula["tbb"].opt_lib}, some reference #{Formula["tbb"].lib}, some reference ${lib}.  Should I get the directories from
a) the shapelib formula using opt_lib, opt_include
b) the shapelib formula using lib, include,
c) from opt_lib, opt_include without referencing a formula
d) from lib, include without referencing a formula

**Comment by SMillerDev** [comment]
1) use as many macOS parts as possible 

2) You should use opt_lib. `lib` requires the formula to be linked and that might not always be the case.

**Comment by tsteven4** [comment]
So case c.  No reference to the shapelib formula at all.  Is that correct?

**Comment by SMillerDev** [comment]
> So case c. No reference to the shapelib formula at all. Is that correct?

Case A actually, there is no way to refer to opt_lib without a formula since it refers to something like `/usr/local/opt/php/lib/`

**Comment by tsteven4** [comment]
I think we are done.  In 18 years gpsbabel has never offered this much control over library configuration.

**Review by SMillerDev — DISMISSED** [review]
Thanks for all the work @tsteven4 ! Without contributions like yours it'd be impossible to keep homebrew going with the high standards that users have come to expect from the project. You can feel good knowing that you've made the world a tiny bit better for homebrew users around the world! 👍 🎉

**Comment by SMillerDev** [comment]
Can you squash your commits into the same title as the PR?

**Comment by tsteven4** [comment]
I'm not sure what you want.  I didn't see any preference for squashed commits in the formula cookbook.  I really need to move on to work upstream.

I did check the tmp directory when I built from source and the PR patch applied cleanly.  That PR was started from the tag gpsbabel_1_7_0 the formula uses just to avoid any difficulties.

As it happens I did a squash and merge on the upstream PR, and deleted the branch, but that patch isn't quite as clean because of intervening commits.  If I use the patch from the merge commit one file will be different.  It won't cause problems with the specific build options the formula uses, but it will cause compilation failures with other options.

**Comment by SMillerDev** [comment]
For our workflow we want one commit named `<formula> <version>` just like the name of the pull request is. (This isn't in the cookbook since it's not related to writing the code)

**Comment by tsteven4** [comment]
I still don't know what you mean.  Do you mean you want a single commit patch in https://raw.githubusercontent.com/Homebrew/formula-patches?  The cookbook does say external patches are accepted, and there are lots of examples of commit and PR patches from upstream repos.

**Comment by SMillerDev** [comment]
No, I want this pull request to be a single commit.

**Comment by tsteven4** [comment]
Thanks for the clarification.  Its squashed.

**Review by SMillerDev — APPROVED** [review]
Thanks @tsteven4  ! Without contributions like yours it'd be impossible to keep homebrew going with the high standards that users have come to expect from the project. You can feel good knowing that you've made the world a tiny bit better for homebrew users around the world! 👍 🎉

**Pull request closed without merging**
